### PR TITLE
[Impeller Scene] Add SceneC asset importing

### PR DESCRIPTION
**Pull request by bdero**
Add asset bundling and hot reloading for 3D models.

Sister patch of https://github.com/flutter/engine/pull/38676.

https://user-images.githubusercontent.com/919017/211301023-fab1ef50-a89e-40aa-a569-5a583690afea.mov

### CODE REVIEW

**Review by jonahwilliams — APPROVED** [review]
Wow, perfect!

I think we should have a flag around this that is just controlled by a compile time constant that we can switch off/on for testing. that is probably the simplest path.

```dart
const bool kIs3dSceneSupported = false;
```

And then guard the asset manifest parsing code to avoid loading anything from models unless that is true. All of the rest of the code can remain as is.

**Comment by bernaferrari** [comment]
Hey, I just want to hijack your PR saying this is AMAZING. Congrats!!

File: packages/flutter_tools/lib/src/flutter_manifest.dart [hunk 1]
```diff
@@ -13,6 +13,9 @@ import 'base/user_messages.dart';
 import 'base/utils.dart';
 import 'plugins.dart';
 
+/// Whether or not Impeller Scene 3D model import is enabled.
+const bool kIs3dSceneSupported = false;
```
Comment (bdero) [inline packages/flutter_tools/lib/src/flutter_manifest.dart:17]: @jonahwilliams I hardcoded the flag here and made it so stuff doesn't get added to the models list -- LMK if this is good or if I should maybe wire it up somewhere else.

I see this this kind of thing is done elsewhere, like for `kHotReloadDefault` and `kIconTreeShakerEnabledDefault`.

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/flutter_manifest.dart:17]: SGTM

**Comment by hryndel2** [comment]
Hi! This is really cool!! I have a question: can I play animations on models? I have a bee model that has three animations. Is there a way to play them?

**Pull request merged**